core/chainio/avs_writer.go

```diff
@@ -107,6 +109,18 @@ func (w *AvsWriter) SendAggregatedResponse(
 	taskResponse messages.CheckpointTaskResponse,
 	aggregation messages.MessageBlsAggregation,
 ) (*types.Receipt, error) {
+	// Wait a block if the task TaskCreatedBlock is the same as the current block
+	currentBlock, err := w.client.BlockNumber(ctx)
```

Comment:
The SendAggregatedResponse shalln't be triggered at the first place if conditions for writing aren't met.

I propose extracting this `sendAggregatedResponseToContract`

Reply:
Could be, np with that

Reply:
Done in [bd3ab42](https://github.com/NethermindEth/near-sffl/pull/136/commits/bd3ab42a54b665762050a99dca7fe3bc992656eb)